### ISSUE TITLE
Multiple worker_gateway.authorized_keys passed in as an array results in none being used by the TSA process

### ISSUE THREAD

**Issue by edwardecook** [concourse/concourse-bosh-release#18]
# Bug Report

Hi, we (Redis for PCF) have a concourse deployment that includes both internal workers (which all have the same worker key) and external workers (which have a different worker key). We have recently started work on upgrading from 4.2.3 to 5.0.0 and found that the way in which we are templating in our web authorized keys:
```
properties:
  worker_gateway:
    authorized_keys: [((external_worker_key.public_key)), ((internal_worker_key.public_key))]
```
 does not work despite the suggestion in the spec that it should: https://github.com/concourse/concourse-bosh-release/blob/0265654ac1ae6160fb5f974056e6564a2564a0f5/jobs/web/spec#L1100-L1101

It appears that the issue revolves around the fact that the `env_file_writer` func, https://github.com/concourse/concourse-bosh-release/blob/0265654ac1ae6160fb5f974056e6564a2564a0f5/jobs/web/templates/pre_start.erb#L22-L29 puts values into separate files when given an array and then passes through a comma separated list of files, https://github.com/concourse/concourse-bosh-release/blob/0265654ac1ae6160fb5f974056e6564a2564a0f5/jobs/web/templates/bpm.yml.erb#L840 while the TSA command expects to be passed a single file path: https://github.com/concourse/concourse/blob/c0422a90aed2ff18b8411b9f5dcaf87b784802b8/tsa/tsacmd/command.go#L35

* Concourse version: 5.0.0
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: GCP
* Browser (if applicable): N/A
* Did this used to work? Yes

Note: as a workaround we have moved to passing the authorized keys line separated in a single string which works:
```
properties:
  worker_gateway:
    authorized_keys: |
      ((external_worker_key.public_key))
      ((internal_worker_key.public_key))
```

cc @jplebre

**Comment by cirocosta** [comment]
Hi @edwardecook ,

Thanks for putting the time to describe the issue so well!
Do you think it'd suffice to have an example as we have for `team_authorized_keys`?

https://github.com/concourse/concourse-bosh-release/blob/0265654ac1ae6160fb5f974056e6564a2564a0f5/jobs/web/spec#L1104-L1114

Such an example could perhaps have multiple keys concatenated there as you mentioned.

Wdyt?

thx!

**Comment by edwardecook** [comment]
Hi @cirocosta, I think an example would certainly be reasonable, the issue I would have is that it does feel like a bit of a regression with 5.0.0 as with 4.2.3 passing in an array worked.

**Comment by cirocosta** [comment]
Oh, I see, indeed!

https://github.com/concourse/concourse-bosh-release/blob/f3af9eb66ec42db163a0b2d794608c715d39f176/jobs/tsa/spec#L70-L73

**Comment by cirocosta** [comment]
cc @vito, how do you see these changes being documented? I remember talking about something related the other day, but I don't remember the output of that 😬 Should we document this separately from `concourse-ci.org`?

**Comment by vito** [comment]
Yeah, sorry about the lack of release notes for this. I wonder if we should start giving this repo its own release cycle and its own release notes; I tend to focus on the core feature set for concourse-ci.org - it's also doesn't even mention all our work on the Helm chart. 🤔

**Comment by vito** [comment]
Now that I look at it, this definitely doesn't seem to be working how I intended - as @edwardecook pointed out, the spec description says an array should work, and that description was written fresh for 5.0.

I'm almost certain I had this working at some point, but maybe I had to change it for some other property that required *its* array value to be passed one-by-one. Looking at the other properties, I can't find any though. The only hint I can find is that [`env_file_content`](https://github.com/concourse/concourse-bosh-release/blob/db6f797ab67d6083827d04951742db6bf1176643/tmpl/create_env_files.erb.tmpl#L6-L15) handles array values by line-separating them. It's `env_file_writer` and `env_file_flag` that seem broken.

Probably a good time to backfill tests for all this. There's a start on this in #20.
